packages/iov-bcp-types/src/transactions.ts

```diff
@@ -90,17 +106,35 @@ export interface SwapTimeoutTx extends BaseTx {
   readonly swapId: SwapIdBytes; // pulled from the offer transaction
 }
 
+export interface RegisterBlockchainTx extends BaseTx {
+  readonly kind: TransactionKind.RegisterBlockchain;
+  /** the ID of the blockchain to be registered */
+  readonly blockchainId: ChainId;
+  readonly codecName: string;
+  readonly codecConfig: string;
+}
+
 export interface RegisterUsernameTx extends BaseTx {
   readonly kind: TransactionKind.RegisterUsername;
   readonly username: string;
   readonly addresses: Map<ChainId, Address>;
```

Comment:
Now that you have introduced `ChainAddressPair `, do we still want a `Map` here or `ReadonlyArray<ChainAddressPair>`?

Reply:
I prefer the later one as it better reflects what we get from the server. Working with Maps is not always as convenient as it could be in JS land and the server will take care we have no duplicate keys hopefully. Map only exist for history reasons.